mget.go

```diff
@@ -104,27 +93,28 @@ type mgetDoc struct {
     Index string `json:"_index,omitempty"`
 }
 
-// MGetResponse wraps the functionality of [opensearchapi.Response] by supporting request unmarshalling of the found
-// documents into K
-type MGetResponse[K RoutableDoc] struct {
-    StatusCode int
-    Header     http.Header
-    Docs       []MGetResult[K]
+// MGetResponse wraps the functionality of [opensearchapi.Response] by unmarshalling the reponse body into a
+// slice of MGetResults
+type MGetResponse struct {
+    StatusCode int          `json:"-"`
+    Header     http.Header  `json:"-"`
+    Docs       []MGetResult `json:"docs,omitempty"`
 }
 
-// mgetResponseBody is an intermediary struct for the api response, to individually un marshall the response documents.
-type mgetResponseBody struct {
-    Docs []json.RawMessage `json:"docs"`
+// MGetResult is the individual result for each requested item
+type MGetResult struct {
+    Index       string          `json:"_index,omitempty"`
+    ID          string          `json:"_id,omitempty"`
+    Version     int             `json:"_version,omitempty"`
+    SeqNo       int             `json:"_seq_no,omitempty"`
+    PrimaryTerm int             `json:"_primary_term,omitempty"`
+    Found       bool            `json:"found,omitempty"`
+    Source      json.RawMessage `json:"_source,omitempty"`
+    Error       error           `json:"-"`
 }
 
-// MGetResult is the individual result for each requested item
-type MGetResult[K RoutableDoc] struct {
-    Index       string `json:"_index,omitempty"`
-    ID          string `json:"_id,omitempty"`
-    Version     int    `json:"_version,omitempty"`
-    SeqNo       int    `json:"_seq_no,omitempty"`
-    PrimaryTerm int    `json:"_primary_term,omitempty"`
-    Found       bool   `json:"found,omitempty"`
-    Source      K      `json:"_source,omitempty"`
-    Error       error  `json:"-"`
+// GetSource returns the raw bytes of the document of the MGetResult
+func (m MGetResult) GetSource() []byte {
+    rawSource, _ := m.Source.MarshalJSON()
```

Comment:
`json.RawMessage` is just an alias for `[]byte` so this could be just
```
return []byte(m.Source)
```

Reply:
yea... idk why I'm so bad at type casting in go. I was trying `m.Source.([]byte)` and it was not happy 😅